index.html

```diff
@@ -3624,6 +3624,86 @@ <h3>Reserved Extension Points</h3>
         </p>
 
       </section>
+
+      <section class="normative">
+        <h3>Ecosystem Compatibility</h3>
+
+        <p>
+There are a number of digital credential formats that do not natively use the
+data model provided in this document, but are aligned with a number of concepts
+in this specification. At the time of publication, examples of these digital
+credential formats include
+<a href="https://www.rfc-editor.org/rfc/rfc7519.html">
+JSON Web Tokens</a> (JWTs),
+<a href="https://www.rfc-editor.org/rfc/rfc8392.html">
+CBOR Web Tokens</a> (CWTs),
+<a href="https://www.iso.org/standard/69084.html">ISO-18013-5:2021</a>
+(mDLs),
+<a href="https://hyperledger.github.io/anoncreds-spec/">
+AnonCreds</a>,
+<a href="https://www.ietf.org/archive/id/draft-mcnally-envelope-02.html">
+Gordian Envelopes</a>, and
+<a href="https://www.ietf.org/archive/id/draft-ssmith-acdc-02.html">
+Authentic Chained Data Containers</a> (ACDCs).
+        </p>
+
+        <p>
+If conceptually aligned digital credential formats can be transformed into a
+<a>conforming document</a> according to the rules provided in this section, they
+are considered <em>"compatible with the Verifiable Credentials ecosystem"</em>.
+A <a>conforming document</a> is either a <a>verifiable credential</a> serialized
+as the `application/vc+ld+json` media type or a <a>verifiable presentation</a>
+serialized as the `application/vp+ld+json` media type. Specifications that
+describe how to perform transformations that enable compatibility with
+the Verifiable Credentials ecosystem:
+        </p>
+
+        <ul>
+          <li>
+MUST identify whether the transformation to this data model is one-way-only or
+round-trippable.
+          </li>
+          <li>
+MUST preserve the `@context` values when performing round-trippable
+transformation.
+          </li>
+          <li>
+MUST result in a <a>conforming document</a> when transforming to the data
+model described by this specification.
+          </li>
+          <li>
+MUST specify a registered media type for the input document.
+          </li>
+          <li>
+SHOULD provide a test suite that demonstrates that the specified transformation
+algorithm to the data model in this specification results in
+a <a>conforming document</a>.
+          </li>
+          <li>
+SHOULD ensure that all semantics utilized in the transformed
+<a>conforming document</a> follow best practices for Linked Data. See
```

Comment:
I think that is in contradiction with the rest of the specification. Specifically, I doubt that using `@vocab` alone for is linked data best practices. We should remove that item. Otherwise we would need to revisit the whole JSON-only processing part since it would also mean that an issuer has to follow linked data best practices when issuing VCs which imo would not include using `@vocab` but rather define context and vocab in an appropriate way.

Reply:
The above says SHOULD not MUST. I think it makes sense to say that you SHOULD follow best practices, but you are not required to.

Reply:
We should also provide a link to what it means to follow linked data best practices. Perhaps we have one in the spec already but I couldn't find it.

Reply:
The link is right below this comment:

> See Section <a href="#getting-started"></a>, Section <a href="#extensibility"></a>, and Linked Data Best Practices [[?LD-BP]] for additional guidance.

Reply:
The should seems fine to me, but the value of satisfying the "should" should be visible internal to the current document... In the "value of JSON-LD" section of the core document.

I don't think we have that section yet, so maybe a marker to align with it, when it gets added makes sense... I don't think we should keep adding to the current PR, to accomplish that.

Reply:
I recommend emphasizing linked data best practices more prominently in the specification. It is essential to clarify that these practices should be applied not only during the transformation process but also when generating and authoring the VCs that have the base media type. Considering this, I propose removing it from the current bullet point list as the remainder of the spec already covers it. It seems somewhat limiting that it is included solely in the ecosystem compatibility section.

Reply:
If there is no consensus that this would apply to the remainder of the spec as well, I'd suggest removing this bullet point.

Reply:
> I recommend emphasizing linked data best practices more prominently in the specification. It is essential to clarify that these practices should be applied not only during the transformation process but also when generating and authoring the VCs that have the base media type. Considering this, I propose removing it from the current bullet point list as the remainder of the spec already covers it. It seems somewhat limiting that it is included solely in the ecosystem compatibility section.

This implies that VCs with "private properties", i.e., relying on "issuer-define", would no longer be valid and would require linking the definitions via @context? 

What would this mean for projects that don't, don't need to or cannot express the definitions of private claims via @context?

Reply:
@alenhorvat Exactly, that is part of my point. Either we remove this entirely, or we make it more prominently. But we should not only add that bullet point to the transformation section.

Reply:
> This implies that VCs with "private properties", i.e., relying on "issuer-define", would no longer be valid and would require linking the definitions via `@context`?

No, that's not what we should imply... I don't think anyone is arguing that something that doesn't follow Linked Data best practices to be "invalid"... just "it could be much better." :) 

If it does imply that, we'll need to rewrite it. All it means is that those VCs w/ private properties aren't following LD best practices... which may or may not be appropriate for the use case. A use case where it might be appropriate is a "closed cycle / closed ecosystem use case" -- where everyone knows each other and what each property means and doesn't feel the need to publish a vocabulary or semantics.

> What would this mean for projects that don't, don't need to or cannot express the definitions of private claims via `@context`?

Generally speaking, they can use issuer-defined terms if it makes sense for them to do it. Some people might grumble, but it doesn't violate the specification for them to not document how their system works via the mechanisms provided by the specification.

Reply:
> > This implies that VCs with "private properties", i.e., relying on "issuer-define", would no longer be valid and would require linking the definitions via `@context`?
> 
> No, that's not what we should imply... I don't think anyone is arguing that something that doesn't follow Linked Data best practices is "invalid"... all that is being said is that "it could be much better." :)
> 
> If it does imply that, we'll need to rewrite it. All it means is that those VCs w/ issuer-defined properties aren't following LD best practices... they're taking a JSON-y approach, which may or may not be appropriate for the use case. A use case where it might be appropriate is a "closed cycle / closed ecosystem use case" -- where everyone knows each other and what each property means and doesn't feel the need to publish a vocabulary or semantics. This feature was put into the specification for people that didn't want to publish contexts or vocabularies (for whatever reason).
> 
> > What would this mean for projects that don't, don't need to or cannot express the definitions of private claims via `@context`?
> 
> Generally speaking, they can use issuer-defined terms if it makes sense for them to do it. Some people might grumble, but it doesn't violate the specification for them to not document how their system works via the mechanisms provided by the specification.
> 
> What we're saying here is: You SHOULD document your terms... but you don't have to.

@msporny Why has this to be part of the transformation section? My point is that this has to be added somewhere else. Perhaps to the media types section? It is not specific to the transformation.

Reply:
@awoie @msporny thank you for the clarification. 

If I understand correctly the WG agrees that VCs can be used in both open and closed systems and that in closed systems use of private or issuer-defined claims is acceptable. I believe it is in everyone's interest that the data model is used in both open and closed systems.

I guess then it is just a matter of wording, or? (this and PR1202 clarify many points)

Reply:
> If there is no consensus that this would apply to the remainder of the spec as well, I'd suggest removing this bullet point.

Added an issue to track this here: https://github.com/w3c/vc-data-model/issues/1217

Reply:
@alenhorvat wrote:
> If I understand correctly the WG agrees that VCs can be used in both open and closed systems and that in closed systems use of private or issuer-defined claims is acceptable. 

Yes, correct.

> I believe it is in everyone's interest that the data model is used in both open and closed systems.

This is why the default `@vocab` value going into the base context achieved consensus. Note that some of us still think it's a bad idea to not document your terminology used, even in closed systems, but this compromise was made to ensure the individuals that wanted to not have to define contexts or vocabularies, but use alternate mechanisms, could do so.

> I guess then it is just a matter of wording, or? (this and PR #1202 clarify many points)

Yes, we just need to get the wording right in #1202 and #1203.